File: android/app/src/main/kotlin/com/tripay/logic/InsightsEngine.kt
package com.tripay.logic

import com.tripay.models.Transaction
import com.tripay.models.TransactionCategory

object InsightsEngine {
    fun calculateTotalSpending(transactions: List<Transaction>): Double {
        return transactions.sumOf { it.amount }
    }
    
    fun categorizeSpending(transactions: List<Transaction>): Map<TransactionCategory, Double> {
        return transactions.groupBy { it.category }
            .mapValues { entry -> entry.value.sumOf { it.amount } }
    }
}

File: ios/TriPay/InsightsEngine.swift
import Foundation

struct InsightsEngine {
    static func calculateTotalSpending(transactions: [Transaction]) -> Double {
        return transactions.reduce(0) { $0 + $1.amount }
    }
    
    static func categorizeSpending(transactions: [Transaction]) -> [TransactionCategory: Double] {
        var result: [TransactionCategory: Double] = [:]
        for transaction in transactions {
            result[transaction.category, default: 0] += transaction.amount
        }
        return result
    }
}
